src/controllers/credentials/index.ts

```diff
@@ -24,28 +27,38 @@ export class CredentialsController extends HTMLController {
     const credentials: Credential[] = await this.cloudagent.getCredentials()
     req.log.info('retrieved credentials from a cloudagent %j', credentials)
 
-    const combined = credentials
-      .map((cred: Credential) => {
-        if (cred.credentialAttributes) {
-          return {
-            ...cred,
-            updated_at: new Date(),
-            company_name: cred?.credentialAttributes[0].name === 'company_name' && cred.credentialAttributes[0].value,
-            company_number:
-              cred.credentialAttributes[1].name === 'company_number' && cred.credentialAttributes[1].value,
-          }
-        }
+    const formatted: Credentials = this.format(credentials)
+    for (let i = 0; i < formatted.length; i++) {
+      const [connection]: ConnectionRow[] = await this.db.get('connection', {
+        agent_connection_id: formatted[i].connectionId,
       })
-      .filter((cred) =>
-        search !== '' ? cred?.company_name.toString().toLowerCase().includes(search.toLowerCase()) : true
-      )
-
-    req.log.info('%j', combined, credentials)
-    if (search !== '') {
-      req.log.info('applying search filter... %j', combined) // do some mapping
+      formatted[i].connection = connection
```

Comment:
Be careful here. The Typescript type system is lying to you and `connection` might be undefined. We should probably filter out the credential in that case

Reply:
thanks, just updated to make sure that connections if defined